tests/unit/backend/corpora/api_server/curator/collection/test_dataset.py

```diff
@@ -35,5 +35,76 @@ def test__delete_dataset(self):
+        queries = ["dataset_uuid", "curator_tag"]
+        auth_credentials = [
+            (self.make_super_curator_header, "super", 204),
+            (self.get_auth_headers, "owner", 204),
+            (None, "none", 401),
+            (self.make_not_owner_header, "not_owner", 403),
+        ]
+        for query in queries:
+            for starting_tag in starting_curator_tags:
+                if not starting_tag and query == "curator_tag":
+                    continue
+                for auth, auth_description, expected_status_code in auth_credentials:
+                    with self.subTest(f"{query=}, {starting_tag=}, auth={auth_description}, {expected_status_code=}"):
+                        collection = self.generate_collection(
+                            self.session, visibility=CollectionVisibility.PRIVATE.name
+                        )
+                        dataset = self.generate_dataset(
+                            self.session,
+                            collection=collection,
+                            curator_tag=starting_tag,
+                        )
+                        test_url = f"/curation/v1/collections/{collection.id}/datasets"
+                        if query == "dataset_uuid":
+                            query_string = {query: dataset.id}
+                        elif query == "curator_tag":
+                            query_string = {query: starting_tag}
+                        headers = auth() if callable(auth) else auth
+                        response = self.app.patch(test_url, headers=headers, query_string=query_string, json=test_body)
+                        self.assertEqual(expected_status_code, response.status_code)
+                        self.session.expire_all()
+                        if expected_status_code == 204:
+                            self.assertEqual(dataset.curator_tag, new_tag)
+                        else:
+                            self.assertEqual(dataset.curator_tag, starting_tag)
+
+    def test___invalid_curator_tag(self):
+        collection = self.generate_collection(self.session, visibility=CollectionVisibility.PRIVATE.name)
+        test_tags = ["new.cxg", "no_extention", "uuid"]
+        for tag in test_tags:
+            with self.subTest(tag):
+                dataset = self.generate_dataset(self.session, collection=collection)
+                test_body = {"curator_tag": f"{dataset.id}.h5ad" if tag == "uuid" else tag}
+                test_url = f"/curation/v1/collections/{collection.id}/datasets"
+                query_string = {"dataset_uuid": dataset.id}
+                headers = self.get_auth_headers()
+                response = self.app.patch(test_url, headers=headers, query_string=query_string, json=test_body)
+                self.assertEqual(400, response.status_code)
+                self.session.expire_all()
+                self.assertIsNone(dataset.curator_tag)
+
+    def test___conflict_curator_tag(self):
+        test_tag = "tag.h5ad"
```

Comment:
same issue here, perhaps rename as `test_tag_with_extension`

Reply:
I'm not sure what you're asking me to change here.

Reply:
the name of the variable. `tag.h5ad` is not just a tag, it is a tag + extension.

Reply:
curator_tag = a name + extension

Reply:
the tag name is never used on its own, or given to the user without the extension. They are single unit of information.

Reply:
Ok, I think that makes things a little less compatible in a logical sense because the extension is _not_ included as part of the uuid when the curator uses the uuid as the identifier for the s3 upload, i.e. `../<uuid>.h5ad`

But I agree it is totally viable to have curator_tag = name + extension. So we can do that. But it means we need to update the regex variable names above to make it clear that the extension is part of the curator tag. There should just be one sub-regex at the end of `REGEX`, and it should be named appropriately as the curator tag regex, and it should _include_ the extension.

Reply:
> Ok, I think that makes things a little less compatible in a logical sense because the extension is _not_ included as part of the uuid when the curator uses the uuid as the identifier for the s3 upload, i.e. `../<uuid>.h5ad`

they do have to append it with h5ad.